### PR TITLE
Optional message

### PR DESCRIPTION
**Pull request by FranckTala**
@jhorbulyk  Please review

### CODE REVIEW

File: lib/actions/reassemble.js [hunk 1]
```diff
@@ -6,6 +6,16 @@ let timeHandle;
...
+function randonGUID() {
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
+    // eslint-disable-next-line no-bitwise
+    const r = Math.random() * 16 | 0;
+    // eslint-disable-next-line no-mixed-operators
+    const v = c === 'x' ? r : (r && 0x3 || 0x8);
+    return v.toString(16);
+  });
+}
+
```
Comment (jhorbulyk) [inline lib/actions/reassemble.js:18]: Can you use [a library for this](https://www.npmjs.com/package/uuid)?

File: lib/actions/reassemble.js [hunk 2]
```diff
@@ -43,6 +54,11 @@ async function processAction(msg, cfg) {
     messageData,
   };
 
+  // chcek if messageId exist
+  if (!messageId) {
+    object.messageId = randonGUID();
+  }
+
```
Comment (jhorbulyk) [inline lib/actions/reassemble.js:61]: Can you change line 44-49 to 
```
  const {
    groupSize,
    groupId,
    messageId = randonGUID(),
    messageData,
  } = msg.body;
```
instead?

File: lib/actions/reassemble.js [hunk 3]
```diff
@@ -18,12 +28,13 @@ async function timer(this_) {
 
     // eslint-disable-next-line no-await-in-loop
     await this_.emit('data', messages.newMessageWithBody({
-      groupSize: results.messages.length,
+      groupSize: Object.keys(results.messageIdsSeen).length,
```
Comment (jhorbulyk) [inline lib/actions/reassemble.js:31]: Why is this change needed?

File: lib/actions/reassemble.js [hunk 4]
```diff
@@ -18,12 +28,13 @@ async function timer(this_) {
...
       groupId: results.messages[0].groupId,
       messageData: incomingData,
     }));
     // eslint-disable-next-line no-await-in-loop
     await storage.deleteObjectById(groupList[i]);
+    groupList = [];
```
Comment (jhorbulyk) [inline lib/actions/reassemble.js:37]: Why is this change needed?

**Review by jhorbulyk — CHANGES_REQUESTED** [review]

File: lib/actions/reassemble.js (hunk 3 above)
Comment (FranckTala) [reply lib/actions/reassemble.js:31]: In the previous version, the messageIdSeen was empty, so we used the messages.length array to count the number of element present in the group. But now that the messageIdsSeen(randon gui) is always defined, we used it to know the exact number of elements present in the group.

File: lib/actions/reassemble.js (hunk 2 above)
Comment (FranckTala) [reply lib/actions/reassemble.js:61]: done!

File: lib/actions/reassemble.js (hunk 1 above)
Comment (FranckTala) [reply lib/actions/reassemble.js:18]: Yes Sure!

File: lib/actions/reassemble.js (hunk 4 above)
Comment (FranckTala) [reply lib/actions/reassemble.js:37]: We need to empty the grouplist after each run to make sure all Base cases in the test environment(reassemble.spec.js) can run properly.

File: lib/actions/reassemble.js (hunk 3 above)
Comment (jhorbulyk) [reply lib/actions/reassemble.js:22]: :+1:

File: lib/actions/reassemble.js (hunk 4 above)
Comment (jhorbulyk) [reply lib/actions/reassemble.js]: :+1:

**Review by jhorbulyk — APPROVED** [review]
We'll do QA in our next sprint.

**Comment by stas-fomenko** [comment]
I'm getting 4 different msgs as output like:
```
{"groupSize":1,"groupId":"2021-11-15T16:31:08.947Z","messageData":[1]}
```

and no random guid appears as a key for msg, it's just 1 array with 1 item (incoming message value that I expect from previous step) in it

NOTES:
Input for re-assemble:
```
{
  "messageData": 1, mapped from step 3
  "groupId": "2095-06-25T19:39:55.874Z" mapped from trigger
}
```

(test flow is: simple trigger -> config component where I emit
```
[
  {
    "a": 1
  },
  {
    "a": 2
  },
  {
    "a": 3
  },
  {
    "a": 4
  }
]
```
-> splitter (split into 4 messages))

File: package.json [hunk 5]
```diff
@@ -19,7 +19,8 @@
     "@elastic.io/maester-client": "3.3.0",
     "elasticio-node": "0.0.9",
     "elasticio-sailor-nodejs": "2.6.24",
```
Comment (stas-fomenko) [inline package.json:21]: pls upgrade "elasticio-sailor-nodejs": "2.6.24", ->   "elasticio-sailor-nodejs": "2.6.26",

**Review by stas-fomenko — CHANGES_REQUESTED** [review]
